exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/Metadata.java

```diff
@@ -1696,9 +1712,24 @@ public ColumnMetadata_v3(String[] name, PrimitiveTypeName primitiveType, Object
       return nulls;
     }
 
+    /**
+     * Checks that the column chunk has single value.
+     * Returns true if minValue and maxValue are the same, but not null.
+     * Returns true if minValue and maxValue are null and the number of null values
+     * in the column chunk is greater than 0.
+     *
+     * @return true if column has single value
+     */
     @Override
     public boolean hasSingleValue() {
-      return (minValue !=null && maxValue != null && minValue.equals(maxValue));
+      if (nulls != null) {
+        if (minValue != null) {
+          return minValue.equals(maxValue);
```

Comment:
Same comment as above. I think in both v1 & v3, we should check nulls = 0 ( there is no nulls in the column).